### PR TITLE
feat: support validation_steps in configure_fit() [DET-4529]

### PR DESCRIPTION
**Pull request by rb-determined-ai**

### CODE REVIEW

File: harness/determined/keras/_tf_keras_trial.py [hunk 1]
```diff
@@ -738,6 +738,10 @@ def _launch_evaluate(self) -> Any:
             self.enqueuers.append(enqueuer)
             validation_data = enqueuer.data()
 
+        if isinstance(validation_data, tf.data.Dataset):
+            # Handle validation_steps, which in Keras only applies to tf.data.Datasets.
+            steps = self.context._fit_validation_steps
```
Comment (aaron276h) [inline harness/determined/keras/_tf_keras_trial.py:743]: question: did you determined that keras does not support validation steps for keras sequences from docs or from reading the source code?

Comment (rb-determined-ai) [reply harness/determined/keras/_tf_keras_trial.py:743]: From the docs, admittedly.  Though admittedly, from the docs, it also doesn't support sequences for validation_data at all.

Comment (aaron276h) [reply harness/determined/keras/_tf_keras_trial.py:743]: From the docs it seemed like validation steps would work for sequences, but they could definitely be inaccurate.

**Review by aaron276h — APPROVED** [review]

File: harness/determined/keras/_tf_keras_trial.py (hunk 1 above)
Comment (rb-determined-ai) [reply harness/determined/keras/_tf_keras_trial.py:743]: The docs say about `validation_steps`: `Only relevant if validation_data is provided and is a tf.data dataset`

**Pull request merged**